src/components/PreferencesPage.vue

```diff
@@ -386,12 +386,55 @@
             </tr>
         </tbody>
     </table>
-    <CustomInstanceModal v-if="showmodal" @close="showmodal = !showmodal" />
+    <ModalComponent v-if="showmodal" @close="showmodal = !showmodal">
```

Comment:
I'm not really happy with this solution: It would be better to keep the dialog as its own component and just update the custom instances list of the `PreferencesPage` when `@close` of the modal gets triggered. Just pasting all the code inside here doesn't look like a good solution here!

Reply:
Sure, but the consequent of that would be : 

1. Disabling reactivity that consequently affects user experience -- so everytime you add or remove an instance, you would need to refresh the page because the customInstance variable inside PreferencesPage component and CustomInstanceModal is two different being, so they cannot communicate, thus when the one inside CustomInstanceModal gets updated (add/remove function), the other does not. This also defeats the purpose of single page app, and redirecting (technically it is) page is the last thing you wanna do because vue has to load everything from the beginning instead what we want is to just update a little bit part of the page, thus making it efficient, thats why SPA is good.
So not only this is inefficient, this is also irritating, i tested this myself for about 2 - 3 hours and i get irritated that i have to refresh the page, click the modal button, type it again. its frustating, trust me.

2. Data Redundancy. Remember DRY. We have to initialize 2 custominstance, making things more inefficient.

3. Unnecessary component means more workflow behind the back, we're instantiating a CustomInstanceModal, and that modal is insatantiating ModalComponent, unless its something thats used more than one case, like account login/register modal, its not worth it.


So unless you wanna kill user experience, and their hardware, i'd rather not go back.

Reply:
I also suspect you didn't understand fully the motivation behind the change and just see whats left. This is pretty much what expected when one writes long code. If theres anything that you wanna ask about the change, please dont hesitate to ask beforehand.

Reply:
> Everytime you add or remove an instance, you would need to refresh the page because the customInstance variable inside PreferencesPage component and CustomInstanceModal is two different being, so they cannot communicate, thus when the one inside CustomInstanceModal gets updated (add/remove function), the other does not.

You have emits in Vue, which are suited exactly for this purpose: https://vuejs.org/guide/components/events.html#emitting-and-listening-to-events, a potential solution is to maybe add a method to add an instance in the `PreferencesPage.vue`, while the validation logic would remain in `CustomInstanceModal`.

Reply:
> > Everytime you add or remove an instance, you would need to refresh the page because the customInstance variable inside PreferencesPage component and CustomInstanceModal is two different being, so they cannot communicate, thus when the one inside CustomInstanceModal gets updated (add/remove function), the other does not.
> 
> You have emits in Vue, which are suited exactly for this purpose: https://vuejs.org/guide/components/events.html#emitting-and-listening-to-events, a potential solution is to maybe add a method to add an instance in the `PreferencesPage.vue`, while the validation logic would remain in `CustomInstanceModal`.

done, ty for the help.